expyfun/tests/test_version.py

```diff
@@ -3,58 +3,60 @@
 from os import path as op
 import warnings
 
-from nose.tools import assert_raises, assert_true, assert_equal
+import pytest
 
 from expyfun import (ExperimentController, assert_version, download_version,
                      __version__)
 from expyfun._utils import _TempDir
 from expyfun._git import _has_git
 
 
-def test_version():
+def test_version_assertions():
     """Test version assertions."""
     with warnings.catch_warnings(record=True) as w:
         warnings.simplefilter('always')
-        assert_raises(TypeError, assert_version, 1)
-        assert_raises(TypeError, assert_version, '1' * 8)
-        assert_raises(AssertionError, assert_version, 'x' * 7)
+        pytest.raises(TypeError, assert_version, 1)
+        pytest.raises(TypeError, assert_version, '1' * 8)
+        pytest.raises(AssertionError, assert_version, 'x' * 7)
         assert_version(__version__[-7:])
-    assert_true(all('actual' in str(ww.message) for ww in w))
+    assert all('actual' in str(ww.message) for ww in w)
 
-    for want_version in ('090948e', 'cae6bc3', 'b6e8a81'):  # old, broken, new
+    # old, broken, new
+    for wi, want_version in enumerate(('090948e', 'cae6bc3', 'b6e8a81')):
+        print('Running %s' % want_version)
         tempdir = _TempDir()
         if not _has_git:
-            assert_raises(ImportError, download_version, want_version, tempdir)
+            pytest.raises(ImportError, download_version, want_version, tempdir)
         else:
-            assert_raises(IOError, download_version, want_version,
+            pytest.raises(IOError, download_version, want_version,
                           op.join(tempdir, 'foo'))
-            assert_raises(RuntimeError, download_version, 'x' * 7, tempdir)
+            pytest.raises(RuntimeError, download_version, 'x' * 7, tempdir)
             ex_dir = op.join(tempdir, 'expyfun')
-            assert_true(not op.isdir(ex_dir))
+            assert not op.isdir(ex_dir)
             download_version(want_version, tempdir)
-            assert_true(op.isdir(ex_dir))
-            assert_true(op.isfile(op.join(ex_dir, '__init__.py')))
+            assert op.isdir(ex_dir)
+            assert op.isfile(op.join(ex_dir, '__init__.py'))
             got_fname = op.join(ex_dir, '_version.py')
             with open(got_fname) as fid:
                 line1 = fid.readline().strip()
             got_version = line1.split(' = ')[1][-8:-1]
-            ex = want_version if want_version != 'cae6bc3' else '.dev0+c'
-            assert_equal(got_version, ex,
-                         msg='File {0}: {1} != {2}'.format(got_fname,
-                                                           got_version, ex))
+            ex = want_version
+            if want_version == 'cae6bc3':
+                ex = (ex, '.dev0+c')
+            assert got_version in ex, got_fname
```

Comment:
this line I don't get.  why is `got_fname` here?

Reply:
(i.e., why would the `got_version` ever equal the file path, and why would we want to allow that?)

Reply:
In `assert` statements what follows the comma is a message (it is not creating a tuple), e.g.:
```
>>> assert 'dan' in 'cool', 'he is not'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: he is not
```
In this case it's useful to have the filename printed, especially when using `pytest --pdb`.

Reply:
I've been pwned